tests/unit/browser/webengine/test_webenginetab.py

```diff
@@ -0,0 +1,109 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2018 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Test webenginetab."""
+
+from unittest import mock
+import logging
+
+from PyQt5.QtCore import QObject
+import pytest
+QtWebEngineWidgets = pytest.importorskip("PyQt5.QtWebEngineWidgets")
+QWebEnginePage = QtWebEngineWidgets.QWebEnginePage
+QWebEngineScriptCollection = QtWebEngineWidgets.QWebEngineScriptCollection
+
+from qutebrowser.browser.webengine.webenginetab import _WebEngineScripts
+from qutebrowser.browser.greasemonkey import GreasemonkeyScript
+
+pytestmark = pytest.mark.usefixtures('greasemonkey_manager')
+
+
+class TestWebengineScripts:
+    """Test the _WebEngineScripts utility class."""
+
+    class FakeWidget(QObject):
+        """Fake widget for _WebengineScripts to use."""
+
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.scripts = []
+            self.page = mock.create_autospec(QWebEnginePage)
+            self.scripts_mock = mock.create_autospec(
+                QWebEngineScriptCollection
+            )
+            self.scripts_mock.toList.return_value = self.scripts
+            self.page.return_value.scripts.return_value = self.scripts_mock
+
+    def test_greasemonkey_undefined_world(self, fake_web_tab, caplog):
+        """Make sure scripts with non-existent worlds are rejected."""
+        uut = _WebEngineScripts(fake_web_tab)
```

Comment:
@toofar huh, what's `uut`?

Reply:
Also, this should be a fixture providing a `_WebEngineScripts` object with the `FakeWidget`.

Reply:
uut = Unit Under Test